lib/iris/_shapefiles.py

```diff
@@ -3,241 +3,447 @@
 # This file is part of Iris and is released under the BSD license.
 # See LICENSE in the root of the repository for full licensing details.
 
-# Much of this code is originally based off the ASCEND library, developed in
-# the Met Office by Chris Kent, Emilie Vanvyve, David Bentley, Joana Mendes
-# many thanks to them. Converted to iris by Alex Chamberlain-Clay
-
+from __future__ import annotations
 
 from itertools import product
+import sys
+from typing import Optional
 import warnings
 
+from affine import Affine
+import cartopy.crs as ccrs
 import numpy as np
+import pyproj
+from pyproj import CRS, Transformer
+import rasterio.features as rfeatures
 import shapely
-import shapely.errors
 import shapely.geometry as sgeom
 import shapely.ops
 
-from iris.warnings import IrisDefaultingWarning, IrisUserWarning
+import iris
+from iris.exceptions import IrisError
+from iris.warnings import IrisUserWarning
+
+if "iris.cube" in sys.modules:
+    import iris.cube
+if "iris.analysis.cartography" in sys.modules:
+    import iris.analysis.cartography
```

Comment:
This is unprecedented. Something larger is wrong if these modules are not available.

Reply:
I can't remember anymore why I added this (!) so it's now removed.